WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -473,6 +486,37 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForGeneralSettingsIn
             [self.languageTextCell setTextValue:name];
             return self.languageTextCell;
         }
+        case SiteSettingsGeneralTimeZone:
+        {
+            NSString *timeZoneString;
+            if (self.blog.settings.timeZoneString != nil) {
+                timeZoneString = self.blog.settings.timeZoneString;
+            } else {
+                // timeZoneString is nil, so that means we can't fetch settings for this blog
+                // so instead we'll use the blog options to get the timezone
+                timeZoneString = [self.blog getOptionValue:@"timezone"];
+            }
+            if ([timeZoneString isEqualToString:@""]) {
+                NSNumber *gmtOffset;
+                if (self.blog.settings.gmtOffset != nil) {
+                    gmtOffset = self.blog.settings.gmtOffset;
```

Comment:
There seems to be an issue with this logic... I just had my site set to `Asia/Jakarta`, then switched it to `UTC +4:30`. The blog's `gmtOffset` property is `0` so we're displaying "UTC + 0" in the cell, but `getOptionValue:@"gmt_offset"` correctly returns `4.5`. But we're not using it because `gmtOffset` is != nil. Perhaps we need to check the optionValue first?